core/src/epicli/data/common/ansible/playbooks/roles/vault/files/vault/configure-vault.sh

```diff
@@ -0,0 +1,329 @@
+#!/usr/bin/env bash
+# Description: This script configures Hashicorp Vault to be used with Epiphany
+# You can find more information in Epiphany documentation in HOWTO.md
+# TODO: Revoke root token
+# TODO: Add configurable log paths
+# TODO: Make devweb-app policy and role configurable (function integrate_with_kubernetes)
+# TODO: Make Helm chart location configurable (function configure_kubernetes)
+
+HELP_MESSAGE="Usage: configure-vault.sh -c SCRIPT_CONFIGURATION_FILE_PATH -a VAULT_IP_ADDRESS -p {http|https}"
+
+function print_help { echo "$HELP_MESSAGE"; }
+
+function log_and_print {
+    local string_to_log="$1";
+    echo "$(date +"%Y-%m-%d-%H:%M:%S"): $string_to_log" | tee -a /opt/vault/logs/configure_vault.log;
+}
+
+function exit_with_error {
+    local string_to_log="$1";
+    log_and_print "ERROR: $string_to_log";
+    exit 1;
+}
+
+function check_status {
+    local exit_code="$1";
+    local success_message="$2";
+    local failure_message="$3";
+    if [ "$exit_code" = "0" ] ; then
+        log_and_print "$success_message";
+    else
+        exit_with_error "$failure_message Exit status: $exit_code";
+    fi
+}
+
+function initialize_vault {
+    local init_file_path="$1";
+    log_and_print "Checking if Vault is already initialized...";
+    vault status -format json | grep -q '"initialized": true';
+    local command_result=( "${PIPESTATUS[@]}" );
+    if [ "${command_result[0]}" = "1" ] ; then
+        exit_with_error "There was an error during checking status of Vault.";
+    fi
+    if [ "${command_result[1]}" = "0" ] ; then
+        log_and_print "Vault is already initialized.";
+    elif [ "${command_result[1]}" = "1" ] ; then
+        log_and_print "Initializing Vault...";
+        vault operator init > "$init_file_path";
+        check_status $? "Vault initialized." "There was an error during initialization of Vault.";
+    fi
+}
+
+function unseal_vault {
```

Comment:
why is that code repeated with unseal-vault.sh script?

Reply:
Because I am not sure if this code written in bash script with using vault CLI is a good idea. Maybe approach with using API directly with some language with better error handling should replace main script. Anyway for MVP version I think that this can be left as it is and is not worth to spend more time on it.

Reply:
Agreed with Pawo, some of that stuff is better written in Python but for MVP the shell script should be fine.

Reply:
Understood.